### ISSUE TITLE
Disable LDPI for oppia app

### ISSUE THREAD

**Issue by rt4914** [oppia/oppia-android#4109]
LDPI is generally used by watches and we should be targeting MDPI and above only.

Disable LDPI for oppia android app.

For the solution, see: https://github.com/oppia/oppia-android/issues/4109#issuecomment-1100236130.

**Comment by BenHenning** [comment]
So a problem that #4136 exposed when trying to implement this is that there actually isn't a good way to disable LDPI. In particular, per https://developer.android.com/guide/topics/manifest/compatible-screens-element:
- Android requires apps to define an allowlist for display densities (this isn't ideal since we only want to remove one: LDPI; an allowlist means we might miss new densities that could be added in the future).
- The manifest entry only affects how Play Store determines device compatibility; it otherwise doesn't affect whether the app can actually be installed on an LDPI device (such as in the case of a universal APK).
- The documentation indicates that apps are intended to support all displays (generally by default, but this is rarely the case; apps usually need to have some special casing for certain display densities).

Given the above, I'm actually inclined not to go ahead with disabling LDPI and instead ship it "as-is" with no explicit support until we actually gather data and user feedback that people actually need LDPI (at which point we can decide from a product perspective how to proceed). We otherwise would close LDPI-specific issues and not make any explicit effort toward supporting it.

@rt4914 from a CLAM perspective, do you have any concerns with this?

@isalooo and @srushtirk I'm also interested in your thoughts here. I don't think we can easily query who would be affected by LDPI yet (since it's predominately used by Android watches), but this is an awkward situation where we don't have a good way to reasonably opt out and there's a non-trivial technical maintenance cost required here.

**Comment by deepanshu731** [comment]
any updates @rt4914 @isalooo @srushtirk ??

**Comment by rt4914** [comment]
@BenHenning @deepanshu731 I recently tried the same thing and based on my experience I do agree that what Ben has suggested sounds good. 

Just one suggestion: We should do one full audit again and make sure that it does work fully (even if UI is bad).

**Comment by BenHenning** [comment]
Another option might be to show a UI prompt specifically for LDPI devices upon opening the app that it isn't available. I can try to figure out the list of possible LDPI devices that the app "could be" installed on if that helps.

@isalooo and @srushtirk it would be nice to get your thoughts on this.

**Comment by isalooo** [comment]
@srushtirk and I will discuss later today and provide an update shortly.

**Comment by srushtirk** [comment]
Agree with Ben's approach and reasoning on not disabling LDPI. However, can we have a list of all the affected devices (if any non-watches) so that we can share with the partners to make them aware of this issue - and they can avoid using those devices for user study and research temporarily until a permanent solution is implemented. Additionally like Ben mentioned, an UI prompt for users will be very helpful for the ones who do end up using these devices, so that they avoid issues in the middle of learning.

**Comment by BenHenning** [comment]
There are 53 devices per the Android device catalog that our app supports and would be classified as under MDPI (i.e. LDPI). Once we eliminate KitKat support from that list, only 23 devices remain (see [this sheet](https://docs.google.com/spreadsheets/d/1ZnoLGZp9sGRCVG3IMh7pu6XixzMgAK8jVCupB1cn69I/edit?usp=sharing)).

I do want to clarify one thing @srushtirk, I think we should be making a final decision here, not a stopgap. That being said, per https://support.google.com/googleplay/android-developer/answer/7353455?hl=en#zippy= we might actually be able to just exclude these devices. We may need to add others over time, but the list *is* fairly small. Would that be viable?

**Comment by srushtirk** [comment]
Makes sense, these devices mostly look like tablets in vehicles and also a very small number as compared to widely used devices in the community. Let's exclude these devices as a final solution.

**Comment by BenHenning** [comment]
Based on that @srushtirk what should we do if a new LDPI device is introduced in the catalog (which happens from time to time). It may go unnoticed for a while; is it fine that the app experience may be broken for these situations or should we still add a notice for such users?

**Comment by srushtirk** [comment]
Please add a notice for the users. This can avoid any unknown surprises for our users and can also help us build trust by proactively communicating this issue to them.

**Comment by BenHenning** [comment]
Can you provide details on what that flow should be @srushtirk?

**Comment by srushtirk** [comment]
I'd need to understand the user journey better and would like to schedule some time to discuss this. 
I see that this is an issue for the GA version. When is the latest that you need a decision on this?

**Comment by BenHenning** [comment]
@srushtirk It's not imminently blocking, but it'd be good to get a decision on this in the next few weeks if possible.

**Comment by BenHenning** [comment]
@seanlip this is a pending product question that needs resolution--can we catalog it?

**Comment by seanlip** [comment]
Yup, thanks for flagging! This was actually already on my list to chat with you about when we next meet -- let's discuss it then.

**Comment by seanlip** [comment]
Based on discussion with @BenHenning, let's do the following:

- For now, individually prohibit the LDPI devices (like watches) that we're aware of from installing the app. (If a new LDPI device comes out subsequently, we can add it to the denylist.)
- For such cases, we'll add a notice in the app along the lines of "Oppia is not supported on small-screen devices. Please use a different device to install Oppia." I will work with the UXW team to confirm/adjust this messaging; assigning the issue to myself in the meantime.

**Comment by seanlip** [comment]
Actually, I'm going to just suggest this message, since this is more of a tangential use case which shouldn't require a lot of additional deliberation on our part. Let's go with:

"Sorry, Oppia is not supported on small-screen devices. Please use another device to install Oppia."

**Comment by BenHenning** [comment]
We're expecting that basically no one is going to reasonably attempt to install Oppia on an LDPI device today, so we can deal with it later (after launching to GA).
